xbmc/video/VideoDatabase.cpp

```diff
@@ -6216,7 +6216,12 @@ bool CVideoDatabase::GetItems(const std::string &strBaseDir, VIDEODB_CONTENT_TYP
   if (StringUtils::EqualsNoCase(itemType, "movies") && (mediaType == VIDEODB_CONTENT_MOVIES || mediaType == VIDEODB_CONTENT_MOVIE_SETS))
     return GetMoviesByWhere(strBaseDir, filter, items, sortDescription);
   else if (StringUtils::EqualsNoCase(itemType, "tvshows") && mediaType == VIDEODB_CONTENT_TVSHOWS)
-    return GetTvShowsByWhere(strBaseDir, filter, items, sortDescription);
+  {
+    Filter filter2(filter);
+    if (!CSettings::GetInstance().GetBool(CSettings::SETTING_VIDEOLIBRARY_SHOWEMPTYTVSHOWS))
+      filter2.AppendWhere("totalCount IS NOT NULL AND totalCount > 0");
```

Comment:
Maybe it would be better to introduce a URL option like `showempty` or `hideempty` which could be appended to the base URL and then the actual SQL implementation could be hidden in `CVideoDatabase::GetFilter()` otherwise we end up with the same SQL query hacks in different code locations.